### PR TITLE
HS-280: Improve routing in NavigationRail

### PR DESCRIPTION
**Pull request by geraldhumphries**
## Description

This wraps the `FeatherAppRailNavItem` components with a `router-link`. There was a good amount of duplicate code so I moved it to its own component.

Pages will no longer fully refresh when clicking things in the navbar. Layout and style should be exactly the same.

I was able to debug the reason text is black in the sidebar, it appears to be a FeatherDS bug caused by styles depending on CSS import order. I didn't include a fix for that in this PR.

I've also removed the redirectUri-based theming for Keycloak, it now uses localStorage.

## Jira link(s)
- https://issues.opennms.org/browse/HS-280

## Checklist
* [ ] Follows Horizon Stream's [development guidelines.](https://github.com/OpenNMS/horizon-stream/wiki/Development-Guidelines)
* [ ] Appropriate reviewer(s) have been selected.
* [ ] Jira issue(s) have been updated to "In Review".
* [ ] Includes [appropriate tests.](https://github.com/OpenNMS/horizon-stream/wiki/Test-Strategy)
* [ ] Documentation has been updated as necessary.
* [ ] Notify devops of changes to the Charts

### CODE REVIEW

File: ui/src/components/Layout/NavigationRail.vue [hunk 1]
```diff
@@ -89,8 +80,9 @@ import Discovery from '@featherds/icon/action/Search'
 import Monitoring from '@featherds/icon/hardware/MinionProfiles'
 import Cycle from '@featherds/icon/action/Cycle'
 import Warning from '@featherds/icon/notification/Warning'
+import NavigationRailNavItem from '@/components/Layout/NavigationRailNavItem.vue'
 
-const Icons = markRaw({
```
Comment (chinh-cle) [inline ui/src/components/Layout/NavigationRail.vue:93]: I think it's good to have `markRaw` for optimization

**Review by chinh-cle — APPROVED** [review]
LGTM

**Review by mikewrosey — APPROVED** [review]
Good to have that working smoother

**Comment by geraldhumphries** [comment]
Rerequested review because I pushed another change to simplify the Keycloak theming.

**Review by chinh-cle — APPROVED** [review]
LGTM

**Pull request merged**